lib/models/per-bundle-addon-cache/addon-proxy.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+/**
+ * Returns a proxy to a target with specific handling for the
+ * `parent` property, as well has to handle the `app` property;
+ * that is, the proxy should maintain correct local state in
+ * closure scope for the `app` property if it happens to be set
+ * by `ember-cli`. Other than `parent` & `app`, this function also
+ * proxies _almost_ everything to `target.targetInstance` with a few
+ * exceptions: we trap & return `[]` for `addons`, and we don't return
+ * the original `included` (it's already called on the "real" addon
+ * by `ember-cli`).
+ *
+ * Note: the target is NOT the per-bundle cacheable instance of the addon. Rather,
+ * it is a cache entry POJO from PerBundleAddonCache.
+ *
+ * @method getAddonProxy
+ * @param targetCacheEntry the PerBundleAddonCache cache entry we are to proxy. It
+ * has one interesting property, 'targetInstance' (though callers here can use the
+ * property 'instance' to get it). The targetInstance is the real addon instance
+ * the proxy is forwarding calls to.
+ * @param parent the parent object of the proxy being created (the same as
+ * the 'parent' property of a normal addon instance)
+ * @return Proxy
+ */
+function getAddonProxy(targetCacheEntry, parent) {
+  let _app;
+  let addonProxy = new Proxy(targetCacheEntry, {
+    get(targetCacheEntry, property) {
+      if (property === 'parent') {
+        return parent;
+      }
+
+      if (property === 'app') {
+        return _app;
+      }
+
+      // keep proxies from even trying to set or initialize addons
+      if (property === 'initializeAddons') {
+        return undefined;
+      }
+
+      // See the {@link index.js} file for a discussion of why the proxy 'addons'
+      // property returns an empty array.
+      if (property === 'addons') {
+        return [];
+      }
+
+      // allow access to `instance`
+      if (property === 'targetInstance') {
```

Comment:
I changed the implementation in `ember-engines` to just check if `PerBundleAddonCache` exists on the project then call a method (getAddonAddons) to do what's needed, rather than having to export this name/symbol.